### PR TITLE
Update CMake Configuration

### PR DESCRIPTION
**Pull request by vvolkl**
Make python bindings optional

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -60,8 +62,8 @@ find_package(Eigen3 REQUIRED)
 include_directories(${EIGEN3_INCLUDE_DIR})
 
 if (tricktrack_logger)
-  find_package(spdlog REQUIRED)
-  include_directories(${SPDLOG_INCLUDE_DIR})
```
Comment (JavierCVilla) [inline CMakeLists.txt:64]: Will be `spdlog` always included as part of `tricktrack` or could it be taken as an external package?

File: build.sh [hunk 2]
```diff
@@ -1,13 +1,10 @@
-git submodule init
-git submodule update
 mkdir build install
-git apply --stat python2.7.patch
+git submodule add https://github.com/pybind/pybind11 python/pybind11
```
Comment (JavierCVilla) [inline build.sh]: Regarding `pybind11`, IIUD running this `build.sh` you will get the `pybind11` code you need to compile files inside the `python` directory. However, this `build.sh` may not be run using other package manager tools like (`lcgcmake` or `spack`), where `pybind11` should be defined as an external dependency, so I think `CMakeLists.txt` should also look for this package in order to be found without explicitly defining `LD_LIBRARY_PATH` or `PYTHONPATH`.

What do you think? I don't really know how was this `pybind11` dependency handled before, so I'm thinking out loud.

File: init.sh [hunk 3]
```diff
@@ -1,2 +1,4 @@
 #on lxplus / machines with cvmfs
 source /cvmfs/fcc.cern.ch/sw/views/releases/0.9.1/x86_64-slc6-gcc62-opt/setup.sh
```
Comment (JavierCVilla) [inline init.sh]: Does `tricktrack` have any dependency with FCC software? Or is it just to prepare a LCG-based environment?

**Review by JavierCVilla — COMMENTED** [review]
@vvolkl Here are some comments having in mind how `tricktrack` could be installed and distributed using package manager tools like LCGCMake or Spack.

File: CMakeLists.txt (hunk 1 above)
Comment (vvolkl) [reply CMakeLists.txt:64]: Could for sure be taken as an external package! I didn't consider this option to be used when installing tricktrack with package manager tools though, is more of an extra for the user / developer.

**Comment by vvolkl** [comment]
Thanks for the comments Javier! The dependencies are really optional in the sense I don't expect them to be installed in LCG releases, but may be very helpful for developers. I tried to address this issue by adding calls to find_package, but fall back to using the versions distributed with tricktrack if not found. Especially for the python bindings the idea is to get a good quick start in tricktrack with a simplified interface, so it makes sense to have a "one build script takes care of everything" approach.

**Comment by JavierCVilla** [comment]
@vvolkl Fine for me, just two more comments:
- If `pybind11` is optional I would just ensure that `tricktrack_python` is always `OFF` by default. 
- Maybe it would be worth to update the docs considering these options. 

Other than that, looks good to me :)

File: CMakeLists.txt [hunk 4]
```diff
@@ -60,8 +62,10 @@ find_package(Eigen3 REQUIRED)
...
+  find_package(spdlog)
+  if (NOT spdlog_FOUND) 
+    include_directories(include/spdlog)
+  endif()
```
Comment (JavierCVilla) [inline CMakeLists.txt:68]: @vvolkl Would you need to put `include_directories(${SPDLOG_INCLUDE_DIR})` back again if `spdlog_FOUND` ?

Comment (vvolkl) [reply CMakeLists.txt:68]: Ah, completely right, I'll patch this!

**Pull request merged**